breathe/project.py

```diff
@@ -172,9 +172,12 @@ def update(
             build_dir
             ):
 
+        # Provide reasonable defaults for domain_by_extension mapping. Can be overridden by users.
+        default_domain_by_extension = {'py': 'py'}
```

Comment:
Yeah, `breathe_domain_by_extension` *is* somewhat cleaner although it might require some user intervention. I've changed it as you suggested in https://github.com/michaeljones/breathe/commit/1142a3a9b6787e3e9f01af567992452c2edbe235.